portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,11 +80,56 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty(COMMENT_TEXT, commentText);
     commentEntity.setProperty(TIME_STAMP, timeStamp);
 
+    // TODO: Remove print statement and store image URL in database.
+    System.out.println(getUploadedFileUrl(request, /* formInputElementName= */ "image").get());
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect(REDIRECT_URL);
   }
+
+  /** 
+   * Returns a URL that points to the uploaded file, or an empty optional
+   * if the user didn't upload an image file. 
+   */
+  private Optional<String> getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (local server)
```

Comment:
Concerning the comment you left: dev server is the local server when testing. Changed from dev to local. The parentheses comments are not complete sentences and I am not sure what they are trying to indicate. If they are important then they should be made into full sentences or else removed.